src/Util/CliLoginHelper.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace ItkDev\OpenIdConnectBundle\Util;
+
+use ItkDev\OpenIdConnectBundle\Exception\TokenNotFoundException;
+use Symfony\Component\Uid\Uuid;
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+ * Helper class for CLI login.
+ */
+class CliLoginHelper
+{
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $cache;
+
+    /**
+     * @var string
+     */
+    private $itkNamespace;
+
+    public function __construct(CacheItemPoolInterface $cache)
+    {
+        $this->cache = $cache;
+        $this->itkNamespace = 'itk-dev-cli-login';
+    }
+
+    /**
+     * Creates login token for CLI login.
+     *
+     * @param string $username
+     * @return string
+     */
+    public function createToken(string $username): string
+    {
+        $encodedUsername = $this->encodeKey($username);
+        $token = Uuid::v4()->toBase32();
+
+        // Add username => token to make sure no username has more than one token
+        $revCacheItem = $this->cache->getItem($encodedUsername);
+        if ($revCacheItem->isHit()) {
+            return $revCacheItem->get();
+        }
+        $revCacheItem->set($token);
+        $this->cache->save($revCacheItem);
+
+        // Add token => username
+        $cacheItem = $this->cache->getItem($token);
+        $cacheItem->set($encodedUsername);
+        $this->cache->save($cacheItem);
+
+        return $token;
+    }
+
+
+    /**
+     * Gets username from login token.
+     *
+     * @param string $token
+     * @return string|null
+     * @throws TokenNotFoundException
+     */
+    public function getUsername(string $token): ?string
```

Comment:
You should do try/catch around the cache calls

Reply:
Added `try/catch` and suppressed `Psr\Cache\InvalidArgumentException` catch in psalm configuration